### ISSUE TITLE
About CRF Transition Matrix Initialization 

### ISSUE THREAD

**Issue by floydluo** [jiesutd/NCRFpp#68]
Hi,

I have a question about CRF's initialization in module`crf.py`.

This content is cited from line 39 to line 43, where `START_TAG` is -2 and `STOP_TAG` is -1.

``` python
        init_transitions = torch.zeros(self.tagset_size+2, self.tagset_size+2)
        init_transitions[:,START_TAG] = -10000.0
        init_transitions[STOP_TAG,:] = -10000.0
        init_transitions[:,0] = -10000.0
        init_transitions[0,:] = -10000.0
```
It is clear that `init_transitions[:,START_TAG]` and `init_transitions[STOP_TAG,:]` are set as -10000. That indicate that `any tag` to `start_tag` is almost impossible and `stop_tag` to `any tag` is almost impossible.

But I couldn't figure out why you set `init_transitions[:,0]` and `init_transitions[0,:]` to -10000.0? It seems that you don't allow `any tag` to be `tags[0]`, as well as not `tags[0]` to be `any tag`.

Best,

Junjie

**Comment by jiesutd** [comment]
Yes, you are right for the START_TAG and STOP_TAG.

For the tag[0] setting, this is because of the id 0 is already been taken in default in [alphabet.py](https://github.com/jiesutd/NCRFpp/blob/138955ff67bff229765ae0cd0fe29984ef175c40/utils/alphabet.py#L26) ,  so the labels actually start from id 1.
